src/components/search/GlobalSearchField.js

```diff
@@ -3,139 +3,485 @@
+            setDataSearchQueryEnabled(false);
+            setOptions([]);
         }
+        setSearchTerm(value);
     };
+
     const handleFilterChange = (event) => {
         setFilter(event.target.value);
     };
+
+    useEffect(() => {
+        if (!open) {
+            setOptions([]);
+        }
+    }, [open]);
+
+    useEffect(() => {
+        if (searchTerm && searchTerm.length > 2) {
+            const userHomeDir = bootstrapCache?.data_info.user_home_path + "/";
+            const dataQuery = getDataSimpleSearchQuery(
+                searchTerm,
+                userHomeDir,
+                ROWS,
+                PAGE
+            );
+            setDataSearchKey([DATA_SEARCH_QUERY_KEY, { query: dataQuery }]);
+
+            setAppsSearchKey([
+                APPS_SEARCH_QUERY_KEY,
+                {
+                    rowsPerPage: ROWS,
+                    orderBy: appFields.NAME.key,
+                    order: constants.SORT_ASCENDING,
+                    page: PAGE,
+                    search: searchTerm,
+                },
+            ]);
+
+            setAnalysesSearchKey([
+                ANALYSES_SEARCH_QUERY_KEY,
+                {
+                    rowsPerPage: ROWS,
+                    orderBy: analysisFields.START_DATE.key,
+                    order: constants.SORT_DESCENDING,
+                    page: PAGE,
+                    filter: getAnalysesSearchQueryFilter(searchTerm),
+                },
+            ]);
+            // console.log("Final Query to submit =>" + JSON.stringify(dataQuery));
+            switch (filter) {
+                case t("data"):
+                    setDataSearchQueryEnabled(true);
+                    setAppsSearchQueryEnabled(false);
+                    setAnalysesSearchQueryEnabled(false);
+                    break;
+
+                case t("apps"):
+                    setDataSearchQueryEnabled(false);
+                    setAppsSearchQueryEnabled(true);
+                    setAnalysesSearchQueryEnabled(false);
+                    break;
+
+                case t("analyses"):
+                    setDataSearchQueryEnabled(false);
+                    setAppsSearchQueryEnabled(false);
+                    setAnalysesSearchQueryEnabled(true);
+                    break;
+
+                default:
+                    setDataSearchQueryEnabled(true);
+                    setAppsSearchQueryEnabled(true);
+                    setAnalysesSearchQueryEnabled(true);
+            }
+        }
+    }, [bootstrapCache, filter, searchTerm, showErrorAnnouncer, t]);
+
+    if (analysesSearchError || appsSearchError || dataSearchError) {
+        showErrorAnnouncer(
+            t("searchError"),
+            analysesSearchError || appsSearchError || dataSearchError
+        );
+    }
+
+    const loading = searchingAnalyses || searchingApps || searchingData;
+
+    const renderCustomOption = (option) => {
+        switch (option?.resultType) {
+            case t("data"):
```

Comment:
Unless I'm mistaken, this `resultType` field isn't rendered anywhere, and given that I think we shouldn't be translating it for use in conditionals like this. In particular, it creates more potential for weird bugs, especially if and when we actually start getting things translated. For example, what happens if the server side doesn't have correct language information while doing SSR, but then the client side tries to do react rehydration using a specific language and produces a different result? Or, what happens if we're switching or branching on a string such that multiple things are translated to the same thing and the conditional loses a branch (don't think that's likely here, but anyway)? These sorts of bugs involving the i18n system or language-switching can't happen if these sorts of things that users don't see just use English, or opaque values instead of any particular language (though that can make writing the code harder, of course).

Reply:
I can easily switch to using non translated values but I still don't understand how translated values will cause a problem.

Reply:
When i18n stuff is broken, our app becomes unusable. For example, the drop down for the filter won't show proper values.

Reply:
One last thing ..our default language is set to `en` in our i18n configs
` defaultLanguage: "en",`

Reply:
Oops I almost forgot, the `resultType` is actually rendered. The `group label` you see in the search result is the `resultType`

Reply:
If it's being displayed, then for sure it should stay as the translated value. We can worry about the potential bugs I'm talking about if they come up later.

Reply:
I'd agree that it seems fragile and risky to use translated strings in conditional checks like this, especially if we have the option to use a known constant instead.

For example, we could create this as an object instead of a simple string: `data.resultType = {label: t("data"), type: DATA}` and then here we could switch on `resultType.type`, and use `groupBy={(option) => option.resultType.label}` for the Autocomplete component. I haven't used that component yet, so I'm only guessing this could work, but I've seen similar logic checks using translated strings for an Autocomplete component before and it bothered me there as well.

Then even if someone made all these keys translate to "X" for some reason, it wouldn't break the logic here (even though it would make the UX less usable).

I think we should at least keep this in mind going forward.